talk/morelanguage/move.tex

```diff
@@ -49,14 +51,14 @@
   \frametitlecpp[11]{Move semantics : the problem}
   \begin{exampleblock}{Another potentially non efficient code}
     \begin{cppcode*}{}
-      std::vector<int> vrandom(unsigned int n) {
-        std::vector<int> result(n);
+      MyVector<int> vrandom(unsigned int n) {
+        MyVector<int> result(n);
```

Comment:
`std::vector` will move nicely here, so I chose a custom type.

Reply:
And copy elision will take care of `MyVector` ...
Maybe we should think about something for the future.

Reply:
Not necessarily. The C++ standard does not guarantee named return value optimization NRVO. A debug build of that program would copy, if `MyVector` does not implement a move constructor.